src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -335,11 +335,25 @@ class DataResourceDataProvider implements IGridDataProvider {
 			if (endIndex > maxRow) {
 				endIndex = maxRow;
 			}
-			let result = this.rows.slice(index, endIndex).map(row => {
+			let result: ICellValue[][] = [];
+			if (includeHeaders) {
+				let columnHeaders: ICellValue[] = columns.map(col => {
+					let headerData: azdata.DbCellValue;
+					headerData = ({
+						displayValue: col.columnName,
+						isNull: false,
+						invariantCultureDisplayValue: col.columnName
+					});
+					return headerData;
+				});
+				result.push(columnHeaders);
+			}
+			this.rows.slice(index, endIndex).map(row => {
```

Comment:
Either use forEach here (instead of map) or just concat the return array onto results (I prefer the second slightly)